cdk_framework/EKS/lib/resource_constructs/general_constructs/test-case-resource-deployment-construct.ts

```diff
@@ -0,0 +1,79 @@
+import { Construct } from 'constructs';
+import { Cluster, FargateCluster } from 'aws-cdk-lib/aws-eks';
+import { NamespaceConstruct } from '../specific_constructs/namespace-construct';
+import { GeneralSampleAppDeploymentConstruct, GeneralSampleAppDeploymentConstructProps } from './general-sample-app-deployment-construct';
+import { GeneralCollectorDeploymentConstruct, GeneralCollectorDeploymentConstructProps } from './general-collector-deployment-construct';
+
+export class TestCaseResourceDeploymentConstruct extends Construct {
+    name: string
+    namespace: Construct
+
+    constructor(scope: Construct, id: string, props: TestCaseResourceDeploymentConstructProps){
+        super(scope, id);
+
+        // namespace deployment
+        const namespaceName = `collector-namespace`
+        const collectorNamespaceConstruct = new NamespaceConstruct(this, 'collector-namespace-construct', {
+            cluster: props.cluster,
+            name: namespaceName
+        })
+
+        // used for when sample app is push mode
+        const deployServices = props.sampleAppMode === 'push'
+        const grpcServiceName = 'collector-grpc'
+        const grpcPort = 4317
+        const udpServiceName = 'collector-udp'
+        const udpPort = 55690
+        const tcpServiceName = 'collector-tcp'
+        const httpPort = 4318
+
+        // sample app deployment
+        const sampleAppLabel = props.sampleAppMode === 'push' ? 'push-mode-sample-app' : 'pull-mode-sample-app'
+        const listenAddressHost = '0.0.0.0'
+        const listenAddressPort = 8080
```

Comment:
Same comment as above, should we move these into the sample app construct as fixed values?